### PR TITLE
Send emails to visitors and prisons when a booking is rejected

### PR DESCRIPTION
**Pull request by threedaymonk**
This sends an email to a prospective visitor when their booking is rejected, and copies it to the prison for their records.

### CODE REVIEW

File: app/models/rejection.rb [hunk 1]
```diff
@@ -11,4 +11,12 @@ class Rejection < ActiveRecord::Base
   belongs_to :visit
 
   validates :reason, inclusion: { in: REASONS }
+
+  def pvo_possible?
```
Comment (SteveMarshall) [inline app/models/rejection.rb]: Can we use this as our chance to stop using `pvo` and `vo` acronyms?

Comment (threedaymonk) [inline app/models/rejection.rb]: If you can capture it in English, I'm all for it!

Comment (threedaymonk) [inline app/models/rejection.rb]: On the other hand, it's actually communicated in the interface as "VO" and "PVO", it's only relevant to prison staff, and it's just a flag/date that gets passed through and has no effect on anything else, so there's an argument that it's in the domain language.

Comment (SteveMarshall) [inline app/models/rejection.rb]: A fair point… Although I might argue that the staff would have that domain knowledge but people working on the codebase might not initially?

Simplistically, `vo` could just be substituted with `visit_order`… But I have no idea what the `p` in `pvo` stands for. Maybe @kyliemulholland knows?

Comment (threedaymonk) [inline app/models/rejection.rb]: The "P" in PVO is "privileged", but it's still an opaque identifier without context.

Comment (kyliemulholland) [inline app/models/rejection.rb]: When a prisoner has an 'enhanced' status, meaning they have above the
standard allocation of visits per month, the extra visits they have are
often exclusively for specific slots. These extra visits are known as PVOs.

Most commonly, in prisons where standard visits happen during the week,
PVOs can be used on weekends. There is no consistent pattern to this
however.

For the purposes of PVB this shouldn't matter, as the staff manage all of
that stuff through NOMIS.

On 24 November 2015 at 10:43, Paul Battley <notifications@github.com> wrote:

> In app/models/rejection.rb
> <https://github.com/ministryofjustice/prison-visits-2/pull/33#discussion_r45719590>
> :
>
> > @@ -11,4 +11,12 @@ class Rejection < ActiveRecord::Base
> >    belongs_to :visit
> >
> >    validates :reason, inclusion: { in: REASONS }
> > +
> > +  def pvo_possible?
>
> The "P" in PVO is "privileged", but it's still an opaque identifier
> without context.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ministryofjustice/prison-visits-2/pull/33/files#r45719590>
> .
>



-- 
*Kylie Mulholland*
Product Manager: Prison Visits Booking
Ministry of Justice | 102 Petty France | London | SW1H 9AJ

Comment (SteveMarshall) [inline app/models/rejection.rb]: @kyliemulholland “manage all of that stuff through NOMIS”… for now! Dun-dun daaa

**Pull request merged**